File: Model.Tests/HtmlExportTests.vb
Imports System.Text.RegularExpressions

<TestClass()>
Public Class HtmlExportTests

    Private Shared Result As String

    <ClassInitialize>
    Public Shared Sub SetUp(testContext As TestContext)
        ' Arrange
        Dim tournament = New Tournament()
        Dim team1 = New Team(1)
        Dim team2 = New Team(2)
        Dim team3 = New Team(3) With {.Withdrawn = True}
        tournament.AddTeam(team1)
        tournament.AddTeam(team2)
        tournament.AddTeam(team3)
        tournament.MatchManager.Generate({team1, team2, team3})
        Dim matchResult = New Result()
        matchResult.AddSetResult(New SetResult() With {.Home = 5, .Guest = 3})
        tournament.MatchManager.SetStatus(2, matchResult)
        Dim testClass = New HtmlExport(tournament)

        ' Act
        Result = testClass.ToString()
    End Sub

    <TestMethod>
    Public Sub HeaderTest()
        StringAssert.StartsWith(Result, "<p>Stand ")
    End Sub

    <TestMethod>
    Public Sub SubtitleTest()
        StringAssert.Contains(Result, " nach 1 von ")
    End Sub

    <TestMethod>
    Public Sub StandingTest()
        StringAssert.Contains(Result, "<td>Team N°1</td>")
    End Sub

    <TestMethod>
    Public Sub GamesOfWithdrawnTeam()
        Dim pattern As New Regex("<td>Team N°3</td>")
        StringAssert.DoesNotMatch(Result, pattern)
    End Sub

    <TestMethod>
    Public Sub FooterTest()
        StringAssert.EndsWith(Result, "</table>")
    End Sub

End Class

File: POFF.Kicker.Tests/HtmlExportTests.cs
using System.Text.RegularExpressions;
using NUnit.Framework;
using POFF.Kicker.Domain;
using POFF.Kicker.Infrastructure;
using POFF.Kicker.View.Model;

namespace POFF.Kicker.Tests;

[TestFixture]
class HtmlExportTests
{
    private static string _result;

    [OneTimeSetUp]
    public static void SetUp()
    {
        // Arrange
        var tournament = new Tournament();
        var team1 = new Team(1);
        var team2 = new Team(2);
        var team3 = new Team(3) { Withdrawn = true };
        tournament.AddTeam(team1);
        tournament.AddTeam(team2);
        tournament.AddTeam(team3);
        tournament.Start(TournamentType.Standard);
        var matchResult = new Result();
        matchResult.AddSetResult(new SetResult() { Home = 5, Guest = 3 });
        tournament.MatchManager.SetStatus(2, matchResult);
        var testClass = new HtmlExport(tournament, ExportType.Games | ExportType.Standings);

        // Act
        _result = testClass.ToString();
    }

    [Test]
    public void StandingTest()
    {
        Assert.That(_result, Does.Contain("<td>Team N°1</td>"));
    }

    [Test]
    public void GamesOfWithdrawnTeam()
    {
        var pattern = new Regex("<td>Team N°3</td>");
        Assert.That(_result, Does.Not.Match(pattern.ToString()));
    }
}